pkg/provisioner/ironic/ironic.go

```diff
@@ -1167,12 +1166,26 @@ func (p *ironicProvisioner) Deprovision() (result provisioner.Result, err error)
 		p.log.Info("cleaning")
 		result.Dirty = true
 		result.RequeueAfter = deprovisionRequeueDelay
+		stepClean, stepCleanOk := ironicNode.CleanStep["step"]
```

Comment:
Ideally we would update the host before setting result.Dirty so that it's clear why we're setting that flag to true.

Reply:
Done